api/api/admin/media_report.py

```diff
@@ -0,0 +1,170 @@
+                        "description",
+                        "has_sensitive_text",
+                    ],
+                },
+            ),
+        ]
+
+    def get_exclude(self, request, obj=None):
+        # ``identifier`` cannot be edited on an existing report.
+        if request.path.endswith("/change/"):
+            return ["media_obj"]
+
+    def get_readonly_fields(self, request, obj=None):
+        if obj is None:
+            return []
+        readonly_fields = [
+            "created_at",
+            "reason",
+            "description",
+            "has_sensitive_text",
+            "media_obj_id",
+        ]
+        # ``status`` cannot be changed on a finalised report.
+        if obj.status != PENDING:
+            readonly_fields.append("status")
+        return readonly_fields
+
+    @admin.display(description="Has sensitive text")
+    def has_sensitive_text(self, obj):
+        """
+        Return `True` if the item cannot be found in the filtered index - which means the item
+        was filtered out due to text sensitivity.
+        """
+        if not self.media_type or not obj:
+            return None
+
+        filtered_index = f"{settings.MEDIA_INDEX_MAPPING[self.media_type]}-filtered"
+        try:
+            search = (
+                Search(index=filtered_index)
+                .query("term", identifier=obj.media_obj.identifier)
+                .execute()
+            )
+            if search.hits:
+                return False
+        except NotFoundError:
+            logging.error(f"Could not resolve index {filtered_index}")
+            return None
+        return True
+
+    def get_other_reports(self, obj):
+        if not self.media_type or not obj:
+            return []
+
+        reports = (
+            self.model.objects.filter(media_obj__identifier=obj.media_obj.identifier)
+            .exclude(id=obj.id)
+            .order_by("created_at")
+        )
+        return reports
+
+    def _get_media_obj_data(self, obj):
+        tags_by_provider = {}
+        if obj.media_obj.tags:
+            for tag in obj.media_obj.tags:
+                tags_by_provider.setdefault(tag["provider"], []).append(tag["name"])
+
+        additional_data = {
+            "other_reports": self.get_other_reports(obj),
+            "identifier": obj.media_obj.identifier,
+            "foreign_landing_url": obj.media_obj.foreign_landing_url,
+            "description": obj.media_obj.meta_data.get("description", ""),
+            "title": obj.media_obj.title,
+            "provider": obj.media_obj.provider,
+            "source": obj.media_obj.source,
+            "creator": obj.media_obj.creator or obj.media_obj.creator_url,
+            "creator_url": obj.media_obj.creator_url,
+            "url": obj.media_obj.url,
+            "tags": tags_by_provider,
+            "reverse_media_url": reverse(
+                f"admin:api_{self.media_type}_change", args=[obj.media_obj.identifier]
+            ),
```

Comment:
Considering how many of these are mappings of properties on `obj.media_obj`, it might be clearer (especially in the template) to pass "media_obj" as a context variable instead. It would make it clearer in the template, for example, where `url`, `creator`, `source`, etc are coming from.

```suggestion
        additional_data = {
            "other_reports": self.get_other_reports(obj),
            "media_obj": obj.media_obj,
            "tags": tags_by_provider,
            "reverse_media_url": reverse(
                f"admin:api_{self.media_type}_change", args=[obj.media_obj.identifier]
            ),
```

Reply:
Thank you for this suggestion, it's much cleaner now!